ada_feeding/scripts/ada_watchdog_listener_node.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+"""
+This module contains a node, ADAWatchdogListener, which listens to the
+watchdog topic and kills itself if the watchdog fails. This is useful
+because launchfiles allow us to perform an action when a node exits,
+so we can use a node like this to terminate the entire launchfile when
+the watchdog fails.
+"""
+
+# Standard imports
+import threading
+from typing import List
+
+# Third-party imports
+from rcl_interfaces.msg import ParameterDescriptor, ParameterType
+import rclpy
+from rclpy.executors import MultiThreadedExecutor
+from rclpy.node import Node
+from std_srvs.srv import SetBool
+
+# Local imports
+from ada_feeding import ADAWatchdogListener
+
+
+class ADAWatchdogListenerNode(Node):
+    """
+    A ROS2 node that listens to the watchdog topic and kills itself
+    if the watchdog fails.
+    """
+
+    def __init__(self) -> None:
+        """
+        Initialize the node.
+        """
+        super().__init__("ada_watchdog_listener")
+
+        # Read the parameter for the rate at which to check the watchdog
+        watchdog_check_hz = self.declare_parameter(
+            "watchdog_check_hz",
+            60.0,  # default value
+            ParameterDescriptor(
+                name="watchdog_check_hz",
+                type=ParameterType.PARAMETER_DOUBLE,
+                description=(
+                    "The rate (Hz) at which to check the whether the watchdog has failed."
+                ),
+                read_only=True,
+            ),
+        )
+
+        # Create a service to toggle this node on-and-off
+        self.toggle_service = self.create_service(
```

Comment:
What's the purpose of this service? I don't think it's bad to have it necessarily, but I do have an issue with the default being that this node is not on and requiring a specific activation.

By default, this node should be on if it is running, and require the user to be _explicit_ when turning it off (either, e.g., setting `watchdog:=false` in the launch file or explicitly calling a service to disable it).

Reply:
So the idea is that since re-taring causes the sensor to stop publishing for 0.75 sec, we could only have the watchdog running when movement is about to occur after re-taring (described in Amal's comment above). However, it's probably a better idea (if possible) to have the ability to toggle the watchdog off and back on around re-taring, rather than having it off as a default. @amalnanavati thoughts?

Reply:
I agree that it should be on by default. Before any MoveTo Behavior, we should do the following:

1. Turn the watchdog listener off.
2. Re-tare the F/T sensor.
3. (Optional) Set F/T thresholds for that motion.
4. Turn the watchdog listener on.

I'll define an idiom that does that.